src/main/java/org/broadinstitute/barclay/help/scanners/DocumentedFeatureScanner.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.barclay.help.scanners;
+
+import jdk.javadoc.doclet.DocletEnvironment;
+import jdk.javadoc.doclet.Reporter;
+import org.broadinstitute.barclay.help.DocWorkUnit;
+import org.broadinstitute.barclay.help.DocletUtils;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.barclay.help.HelpDoclet;
+import org.broadinstitute.barclay.utils.Utils;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.util.ElementScanner14;
+import javax.tools.Diagnostic;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * An {@link ElementScanner14} for finding {@link DocumentedFeature}s to be included in a Barclay doc
+ * task. Returns a set of {@link DocWorkUnit} objects.
+ */
+public class DocumentedFeatureScanner extends ElementScanner14<Void, Void> {
```

Comment:
Shouldn't this be a scanner 17?

Reply:
Yes, but `ElementScanner17` doesn't appear to exist. `ElementScanner14` is annotated with (`@SupportedSourceVersion(RELEASE_17)`. I assume they do this when the new version has no language changes that would warrant a new class, but the doc for these classes is pretty minimal.

Reply:
That makes sense.   I'm surprised 17 didn't have any new elements with the new pattern matching stuff.